src/connectionhealth.cpp

```diff
@@ -103,14 +107,13 @@ void ConnectionHealth::startIdle() {
   m_dnsPingInitialized = false;
   m_dnsPingLatency = PING_TIME_UNSTABLE_SEC * 1000;
 
-  m_dnsPingSender.start();
-  m_dnsPingTimer.start(PING_INTERVAL_IDLE_SEC * 1000);
-
-  // Send an initial ping right away.
-  m_dnsPingTimestamp = QDateTime::currentMSecsSinceEpoch();
-  m_dnsPingSender.sendPing(QHostAddress(PING_WELL_KNOWN_ANYCAST_DNS),
-                           m_dnsPingSequence);
-
+  if (m_dnsPingSender.start()) {
+    m_dnsPingTimer.start(PING_INTERVAL_IDLE_SEC * 1000);
+    // Send an initial ping right away.
+    m_dnsPingTimestamp = QDateTime::currentMSecsSinceEpoch();
+    m_dnsPingSender.sendPing(QHostAddress(PING_WELL_KNOWN_ANYCAST_DNS),
+                             m_dnsPingSequence);
+  }
```

Comment:
Can we add unit tests for this?

Reply:
Let's at least add a ticket for this... Seems a bit pointless, I know, but still.